### PR TITLE
Travis: repair builds on JRuby

### PR DESCRIPTION
**Pull request by olleolleolle**
This PR's fixes the build for all supported build targets.

## TL;DR

- trust Bundler to find correct deps
- configure JRuby's JRUBY_OPTS to avoid warnings

### Attempt: update RubyGems and Bundler

**Versions differ**: The JRuby build failure on master (a few months back) that I inspected was on 1.7.2x, while the current Travis alias to `jruby` now means `9.1.9.0`. Hm, perhaps we'd like this to follow latest releases (by using exact version names).

JRuby 1.7.2x had RubyGems  v2.4.8, while 9.1.9.0 has v2.6.11. 

The `before_install` hook upgrades us to v2.6.12. And, to latest Bundler (v1.15.3), which makes the change which fixed all this possible.

### Attempt: try to make minimal changes to Gemfile

Adding `gem 'rake'` in the Gemfile only served to mutate the lockfile when installing; this is an error in Bundler, so this is a non-starter.

### What does it: keep no lock file for the gem

The change in `Gemfile` is done to rely more on Bundler's feature to manage different platforms and Ruby versions.

- Remove and git ignore the lockfile - to avoid having a "tampered-with" `Gemfile.lock` when we reach the `bundle install` step in Travis, which has the `--deployment` flag set (I even locally encountered https://github.com/bundler/bundler/issues/5050 while debugging this.)
- Explicitly _tell_ Bundler to include "currently running Ruby version" as a gem prerequisite for its "version resolution" mechanism
- Un-pinning versions to Rake for example, allows Bundler's gem finding mechanism to use version-appropriate releases

## After the build runs, avoid the RSpec failure

This gets us to the point where JRuby fails on a test failure about _output_.

JRuby spouts some warnings, which the test does not account for. I first added a commit which weakens the check to see if the output `include`s the desired message instead of `==` the message.

Alternative - non-invasive fix, to avoid having that warning output: `JRUBY_OPTS` that does not include the unsupported setting:
https://github.com/jasmine/jasmine-gem/commit/d16adb62a1746d6c01b41e4c3d024124c7985f01

Source: [Travis CI issue about JRUBY_OPTS](https://github.com/travis-ci/travis-ci/issues/6471#issuecomment-239625123)

### CODE REVIEW

**Comment by grosser** [comment]
some weird shit ... tried a simpler version of that and could not get it to work :/
... but now rbx fails :D 

by default latest is installed 

```
$ bundle --version
Bundler version 1.15.3
$ gem --version
2.6.11
```

... so we don't need to reinstall them ?

**Comment by grosser** [comment]
works without bumping them ... https://github.com/gregorym/bump/pull/76

**Comment by grosser** [comment]
... now #76 is also green on rbx ... idk why :D

**Comment by grosser** [comment]
FYI arguments for checking in Gemfile.lock https://grosser.it/2015/08/14/check-in-your-gemfile-lock/
... so now that the minimal change works, I'd rather use that ...

**Comment by grosser** [comment]
🎉  back to green ... not really sure why this works ... but it's not that bad of a tradeoff ...

**Comment by grosser** [comment]
thx for fixing this ... I'd never had suspected it's a checked in Gemfile.lock :/
